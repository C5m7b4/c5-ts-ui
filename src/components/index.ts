import {
  Button,
  NormalButtonType,
  NormalButtonSize,
  ButtonProps,
} from './Button';

export { Button, NormalButtonType, NormalButtonSize, ButtonProps };
