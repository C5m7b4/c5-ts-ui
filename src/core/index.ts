import Backdrop, { BackdropProps } from './Backdrop';
import Modal, { ModalProps } from './Modal';
import ModalWrapper, { ModalWrapperProps } from './ModalWrapper';

export {
  Backdrop,
  BackdropProps,
  Modal,
  ModalProps,
  ModalWrapper,
  ModalWrapperProps,
};
