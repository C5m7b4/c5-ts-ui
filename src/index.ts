import { Add } from './math';

export { Add };
