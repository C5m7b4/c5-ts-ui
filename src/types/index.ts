export type Theme = 'light' | 'dark' | 'colored';

export type Parsers =
  | 'babel'
  | 'css'
  | 'json'
  | 'flow'
  | 'babel-flow'
  | 'babel-ts'
  | 'typescript'
  | 'json'
  | 'markdown'
  | 'html';
