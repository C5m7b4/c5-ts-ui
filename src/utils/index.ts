export * from './colors';
export * from './reserved';
export * from './fns';
