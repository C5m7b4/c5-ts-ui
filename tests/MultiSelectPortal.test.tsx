import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';

import { MultiSelectPortal } from '../src';
import { data } from './data';

describe('MultiSelectPortal tests', () => {
  const handleCheck = jest.fn();
  const handleSearchChange = jest.fn();
  const amIChecked = jest.fn();

  beforeEach(() => {
    const div = document.createElement('div');
    div.id = 'ms-portal-container';
    document.body.appendChild(div);
  });

  test('should render correcctly', () => {
    const { container } = render(
      <MultiSelectPortal
        data={data}
        keyDescriptor={'select a department'}
        optionText={'value'}
        open={true}
        handleCheck={handleCheck}
        handleSearchChange={handleSearchChange}
        amIChecked={amIChecked}
        search=""
        inputId="inputId"
        id="id"
      />
    );
    expect(container).toMatchSnapshot();
  });

  test('should render correctly without an id', () => {
    const { container } = render(
      <MultiSelectPortal
        data={data}
        keyDescriptor={'select a department'}
        optionText={'value'}
        open={true}
        handleCheck={handleCheck}
        handleSearchChange={handleSearchChange}
        amIChecked={amIChecked}
        search=""
        inputId="inputId"
      />
    );
    expect(container).toMatchSnapshot();
  });

  test('should not draw a portal', () => {
    const { container } = render(
      <MultiSelectPortal
        data={[]}
        keyDescriptor={'select a department'}
        optionText={'value'}
        open={true}
        handleCheck={handleCheck}
        handleSearchChange={handleSearchChange}
        amIChecked={amIChecked}
        search=""
        inputId="inputId"
      />
    );
    expect(container).toMatchSnapshot();
  });

  test('should handle not opening the portal', () => {
    const { container } = render(
      <MultiSelectPortal
        data={[]}
        keyDescriptor={'select a department'}
        optionText={'value'}
        open={false}
        handleCheck={handleCheck}
        handleSearchChange={handleSearchChange}
        amIChecked={amIChecked}
        search=""
        inputId="inputId"
      />
    );
    expect(container).toMatchSnapshot();
  });

  test('should handle checking a box', async () => {
    const { container } = render(
      <MultiSelectPortal
        data={data}
        keyDescriptor={'select a department'}
        optionText={'value'}
        open={true}
        handleCheck={handleCheck}
        handleSearchChange={handleSearchChange}
        amIChecked={amIChecked}
        search=""
        inputId="inputId"
        id="id"
      />
    );

    const check = screen.getByRole('checkbox', { name: /Grocery/ });
    expect(check).not.toBeChecked();
    await fireEvent.click(check);
    expect(check).toBeChecked();
    expect(container).toMatchSnapshot();
  });
});
