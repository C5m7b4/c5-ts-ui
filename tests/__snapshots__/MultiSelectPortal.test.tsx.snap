// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MultiSelectPortal tests should handle checking a box 1`] = `<div />`;

exports[`MultiSelectPortal tests should handle not opening the portal 1`] = `<div />`;

exports[`MultiSelectPortal tests should not draw a portal 1`] = `<div />`;

exports[`MultiSelectPortal tests should render correcctly 1`] = `<div />`;

exports[`MultiSelectPortal tests should render correctly without an id 1`] = `<div />`;
